ibis/expr/operations.py

```diff
@@ -1246,10 +1405,10 @@ def when(self, case_expr, result_expr):
         -------
         builder : CaseBuilder
         """
-        case_expr = as_value_expr(case_expr)
-        result_expr = as_value_expr(result_expr)
+        case_expr = ir.as_value_expr(case_expr)
+        result_expr = ir.as_value_expr(result_expr)
```

Comment:
I think we should get rid of `as_value_expr` too, all the value coercions should be handled by rules.

Reply:
I definitely agree with that.